sdk/src/client/api/block_builder/input_selection/requirement/native_tokens.rs

```diff
@@ -167,16 +155,25 @@ impl InputSelection {
                 // If we created the foundry with this transaction, then we need to add the circulating supply as minted
                 // tokens
                 if initial_creation {
-                    let circulating_supply = output_foundry_simple_ts.circulating_supply();
-
-                    if !circulating_supply.is_zero() {
-                        minted_native_tokens
-                            .add_native_token(NativeToken::new(output_foundry.token_id(), circulating_supply)?)?;
-                    }
+                    *minted_native_tokens.entry(output_foundry.token_id()).or_default() +=
+                        output_foundry_simple_ts.circulating_supply();
                 }
             }
         }
 
-        Ok((minted_native_tokens, melted_native_tokens))
+        (minted_native_tokens, melted_native_tokens)
     }
 }
+
+pub(crate) fn get_native_tokens_diff(
```

Comment:
I think something like `get_native_tokens_remainder()` would be a better name since other entries on the second one are irrelevant. Maybe also rename the params to input/output?

Reply:
fine by me

Reply:
or I can change it to return two maps, one for added and one for removed

Reply:
Du we need the second map somewhere?

Reply:
Well we reverse the call in the native tokens requirement which is kind of like using the second map 🤷‍♀️

Reply:
Ah in this case it's not really remainder, I'll just resolve this :joy: